test/src/org/omegat/filters4/Xliff1FilterTest.java

```diff
@@ -0,0 +1,122 @@
+    @Test
+    public void testParse() throws Exception {
+        List<String> entries = parse(new Xliff1Filter(), "test/data/filters/xliff/filters4-xliff1/en-xx.xlf");
+        assertEquals(7, entries.size()); // the file contains 8 entries but the one with "translate=no" is ignored
+        assertEquals("This is the source text.", entries.get(0));
+        assertEquals("foo", entries.get(1));
+    }
+    
+    @Test // checks that file wihout ID will fail (REQUIRED in the specification)
+    public void testParseMissingId() throws Exception {
+        Exception exception = assertThrows(TranslationException.class, () -> {
+            // Try with a file from filters3 : they are not conform to specification so it should fail
+            parse(new Xliff1Filter(), "test/data/filters/xliff/filters3/file-XLIFFFilter.xlf");        
+        });
+        System.err.println("Parsing file where trans-unit/@id is missing returns: "
+            + exception.getClass() + ": " + exception.getMessage());
+        assertTrue(exception.getMessage().contains("Missing attribute"));
+    }
+
+    @Test
+    public void testBilingual() throws Exception {
+        Map<String,String> result = new HashMap<String,String>();
+        Map<String,String> legacy = new HashMap<String,String>();
+
+        // Test that we correctly read translation
+        parse2(new Xliff1Filter(), "test/data/filters/xliff/filters4-xliff1/en-xx.xlf", result, legacy);
+        assertEquals("bar", result.get("foo"));
+    }
+
+    @Test
+    public void testKey() throws Exception {
+        List<ParsedEntry> entries = parse3(new Xliff1Filter(), "test/data/filters/xliff/filters4-xliff1/en-xx.xlf",
+            java.util.Collections.emptyMap());
+        ParsedEntry firstEntry = entries.get(0);
+        assertEquals(firstEntry.id, "example_01");
+        assertEquals(firstEntry.path, "//interface.po");  // path is built with file name
+        ParsedEntry secondEntry = entries.get(1);
+        assertEquals("foo", secondEntry.source);
+        assertEquals("bar", secondEntry.translation);
+        ParsedEntry lastEntry = entries.get(entries.size() - 1);
+        assertEquals(lastEntry.id, "Added_1");  // next entry Added_2 is ignored as translate=no
+        assertEquals(lastEntry.path, "//interface.po/Group 3");  // path is built with file name and groups              
+    }
+
+    @Test
+    public void testTranslation() throws Exception {
+        Xliff1Filter filter = new Xliff1Filter();
+        filter.translateFile(new File("test/data/filters/xliff/filters4-xliff1/en-xx.xlf"), 
+            outFile, java.util.Collections.emptyMap(), context,
+                new ITranslateCallback() {
+                    public String getTranslation(String id, String source, String path) {
+                        if ("Should translate in result.".equals(source)) {
+                            return "Devrait traduire dans le résultat.";
```

Comment:
Please escape non-ascii character here.

Reply:
OK, I must review  a little bit my tests (especially not finished for XLIFF 2), I will do the change.